ARCs/arc-0018.md

```diff
@@ -0,0 +1,348 @@
+type RoyaltyPolicy {
+    share:              number,  // The share of the payment due, specified in basis points
+    royalty_collector:  string,  // The address that should collect this share of the payment
+    allowed_assets:     number[] // The list of asset ids that are acceptable for payment
+}
+```
+
+### Royalty Enforcer
+
+The Royalty Enforcer is an application that controls the transfer of ASAs subject to the Royalty Policy. 
+
+This is accomplished by exposing an interface defined as a set of [ABI Methods](#abi-methods) allowing a grouped transaction call.
+
+### Royalty Collector
+
+The Royalty Collector is a generic account that could be set to a Single Signature, a Multi Signature, a Smart Signature or even to another Smart Contract. The Royalty Collector is then responsible for the royalty distribution logic, making the Royalty Enforcement Specification more general and composable.
+
+
+### Royalty Asset 
+
+The Royalty Asset is an ASA subject to royalty collection and MUST be created with the the appropriate roles and parameters. 
+
+> Because the protocol does not allow updating a role after it's been deleted, if the creator thinks she may want to modify them later, they must be set to some non-zero address.
+
+
+#### Asset Roles
+
+The Clawback role MUST be set to the Application Address.
+
+> Since the Royalty Enforcer relies on using the Clawback mechanism to perform the transfer the Clawback should NEVER be set to the zero address. 
+
+The Freeze role MUST be set to the Application Address if `FreezeAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to allow an ASA to be Royalty Free after some conditions are met, it should be set to the Application Address 
+
+The Manager role MUST be set to the Application Address if `ManagerAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to update the Freeze role, this should be set to the application address
+
+The Reserve role MAY be set to anything.
+
+#### Asset Parameters
+
+The `DefaultFrozen` MUST be set to true.
+
+
+### Third Party Marketplace
+
+In order to support secondary sales on external markets this spec was designed such that the Royalty Asset may be listed without transferring it from the current owners account.  A Marketplace may call the transfer request as long as the address initiating the transfer has been set as the `authorized_address` through the [offer](#offer) method in some previous transaction by the current owner.
+
+
+### ABI Methods
+
+#### Required
+
+
+##### SetPolicy
+    
+    set_policy(royalty_asset: asset, royalty_receiver: address, royalty_share: uint64, allowed_asset_1: asset, allowed_asset_2: asset, allowed_asset_3: asset, allowed_asset_4: asset)
+
+Sets the Royalty Policy for a given asset. 
+
+The `royalty_asset` is the ASA id of the asset the policy is set for.  If the `royalty_asset` is 0, the policy is defined for _any_ asset that has its freeze and clawback set to the application address. 
```

Comment:
> the policy is defined for _any_ asset that has its freeze and clawback set to the application address.

We should not mandate `FreezeAddress` AND `ClawbackAddress` set to Application Address, but just `ClawbackAddress` since the `FreezeAddress` could be set to `ZeroAddress`.

Reply:
@fabrice102 this would be default if no specific policy is defined for a given Asset ID